pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -40,6 +40,43 @@ type MongoDBSpec struct {
 	// Security configures security features, such as TLS, and authentication settings for a deployment
 	// +optional
 	Security Security `json:"security"`
+
+	// Users specifies the MongoDB users that should be configured in your deployment
+	// +required
+	Users []MongoDBUser `json:"users"`
+}
+
+type MongoDBUser struct {
+	// Name is the username of the user
+	Name string `json:"name"`
+
+	// DB is the database the user is stored in. Defaults to "admin"
+	// +optional
+	DB string `json:"db"`
+
+	// PasswordSecretRef is a reference to the secret containing this user's password
+	PasswordSecretRef SecretKeyReference `json:"passwordSecretRef"`
+
+	// Roles is an array of roles assigned to this user
+	Roles []Role `json:"roles"`
+}
+
+// SecretKeyReference is a reference to the secret containing the user's password
+type SecretKeyReference struct {
```

Comment:
@fabianlindfors currently yes. We could make it possible to reference secrets in different namespaces, but it might be better to start off simpler, and add this functionality when we need it. What do you think?

Reply:
Agreed. We could allow it for TLS soon as well but I don't see a reason to complicate things.